File: src/python/PythonSDK/foundationallm/models/orchestration/message_content_item_base.py
from pydantic import BaseModel, Field
from typing import Optional
from .message_content_item_types import MessageContentItemTypes
from foundationallm.models.constants import AgentCapabilityCategories

class MessageContentItemBase(BaseModel):
    """Base message content item model."""
    
    type: Optional[MessageContentItemTypes] = Field(None, description="The type of message content item.")
    agent_capability_category: Optional[AgentCapabilityCategories] = Field(None, description="The category of capability assigned to the agent.")
    
    class Config:
        use_enum_values = True
        populate_by_name = True
        extra = "forbid"

File: src/dotnet/Common/Models/Orchestration/Response/MessageContentItemBase.cs
﻿using FoundationaLLM.Common.Constants.Agents;
using FoundationaLLM.Common.Constants.Orchestration;
using System.Text.Json.Serialization;

namespace FoundationaLLM.Common.Models.Orchestration.Response
{
    /// <summary>
    /// Base message content item model.
    /// </summary>
    [JsonPolymorphic(TypeDiscriminatorPropertyName = "type")]
    [JsonDerivedType(typeof(TextMessageContentItem), MessageContentItemTypes.Text)]
    [JsonDerivedType(typeof(ImageFileMessageContentItem), MessageContentItemTypes.ImageFile)]
    [JsonDerivedType(typeof(FilePathContentItem), MessageContentItemTypes.FilePath)]
    public class MessageContentItemBase
    {
        /// <summary>
        /// The type of the message content item.
        /// </summary>
        [JsonPropertyName("type")]
        [JsonPropertyOrder(-100)]
        public virtual string? Type { get; set; }

        /// <summary>
        /// The FoundationaLLM agent capability category that generated the message content item.
        /// </summary>
        /// <remarks>
        /// Must be one of the values defined in <see cref="AgentCapabilityCategoryNames"/>.
        /// </remarks>
        [JsonPropertyName("agent_capability_category")]
        [JsonPropertyOrder(-99)]
        public required string AgentCapabilityCategory { get; set; }
    }
}
